src/url.cpp

```diff
@@ -22,8 +22,34 @@ namespace Url
     const std::string Url::QUERY = Url::PCHAR + "/?";
     const std::string Url::FRAGMENT = Url::PCHAR + "/?";
     const std::string Url::USERINFO = Url::UNRESERVED + Url::SUB_DELIMS + ":";
+    const std::string Url::HEX = "0123456789ABCDEF";
     const std::string Url::SCHEME =
         "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789+-.";
+    const std::vector<int> Url::HEX_TO_DEC = {
```

Comment:
Ah, excellent. For some reason I had memories of needing cast each element in certain cases when I had tried to use a `char`, but apparently I misremembered that.